Unit 2.3 LangGraph : Release 11/03/25 (#271)
huggingface/agents-course: units/en/unit2/langgraph/document_analysis_agent.mdx

> # Document Analysis Graph

Reviewer: In this section, since `gpt-4o` is used, maybe adding something in the line of the following (extracted from [here](https://huggingface.co/learn/agents-course/unit2/smolagents/vision_agents))

```
<Tip warning={true}>
The examples in this section require access to a powerful VLM model. We tested them using the GPT-4o API.  
However, <a href="./why_use_smolagents">Why use smolagents</a> discusses alternative solutions supported by smolagents and Hugging Face. If you'd like to explore other options, be sure to check that section.
</Tip>
```

Alfred at your service. As Mr. Wayne's trusted butler, I've taken the liberty of documenting how I assist Mr Wayne with his various documentary needs. While he's out attending to his... nighttime activities, I ensure all his paperwork, training schedules, and nutritional plans are properly analyzed and organized.

Before leaving, he left a note with his week training program. I then took the responsability to come up with a **menu** for tomorrow's meals.

For future such event, let's create a document analysis system using LangGraph to serve Mister Wayne's needs. This system can:

1. Process images document

> 2. Extract text using vision models ( Vision Language Model)

Reviewer: ```suggestion
2. Extract text using vision models (Vision Language Model)
```

> 3. Perform calculations when needed ( to demonstrate normal tools)

Reviewer: ```suggestion
3. Perform calculations when needed (to demonstrate normal tools)
```

4. Analyze content and provide concise summaries
5. Execute specific instructions related to documents

## The Butler's Workflow

The workflow we'll build, follows a structured this schema:

![Butler's Document Analysis Workflow](https://huggingface.co/datasets/agents-course/course-images/resolve/main/en/unit2/LangGraph/alfred_flow.png)

> You can find the following Notebook [Here](https://huggingface.co/agents-course/notebooks/blob/main/unit2/langgraph/agent.ipynb)

Reviewer: ```suggestion
<Tip>
You can follow the code in <a href="https://huggingface.co/agents-course/notebooks/blob/main/unit2/langgraph/agent.ipynb" target="_blank">this notebook</a> that you can run using Google Colab.
</Tip>
```
